lib/web/apiserver_test.go

```diff
@@ -1035,6 +1041,35 @@ func (s *WebSuite) TestResizeTerminal(c *C) {
 	}
 }
 
+func (s *WebSuite) TestTerminalPing(c *C) {
+	ws, err := s.makeTerminal(s.authPack(c, "foo"))
+	c.Assert(err, IsNil)
+	defer ws.Close()
+	closed := false
+
+	done := make(chan struct{})
+	ws.SetPingHandler(func(message string) error {
+		if closed == false {
+			close(done)
+			closed = true
+		}
+
+		err := ws.WriteControl(websocket.PongMessage, []byte(message), time.Now().Add(time.Second))
+		if err == websocket.ErrCloseSent {
+			return nil
+		} else if e, ok := err.(net.Error); ok && e.Temporary() {
+			return nil
+		}
+		return err
+	})
+
+	select {
+	case <-done:
+	case <-time.After(time.Minute * 5):
```

Comment:
nit: 1 minute should be plenty, plus similar change as above?

```suggestion
	case <-time.After(1* time.Minute):
```